cmd/kubeadm/app/preflight/utils.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package preflight
+
+import (
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strings"
+
+	"k8s.io/kubernetes/pkg/util/version"
+)
+
+// GetKubeletVersion is helper function that returns version of kubelet available in $PATH
+func GetKubeletVersion() (*version.Version, error) {
+	kubeletVersionRegex := regexp.MustCompile(`^\s*Kubernetes v((0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)([-0-9a-zA-Z_\.+]*)?)\s*$`)
+
+	out, err := exec.Command("kubelet", "--version").Output()
```

Comment:
I think it would be better to parameterize this, and call `${tmpdir}/kubelet` in tests instead of mocking with $PATH

Reply:
It will be different execution codepath then (in test and in real function). This way it doesn't require from users of this function to add useless parameter every time somebody willing to utilize it. Testing with clean $PATH also allows to remove all possible interferences from host to exec().